cometd-java/cometd-java-server/cometd-java-server-common/src/main/java/org/cometd/server/BayeuxServerImpl.java

```diff
@@ -135,7 +147,7 @@ protected void doStart() throws Exception {
         schedule(new Runnable() {
             @Override
             public void run() {
-                asyncSweep().whenComplete((r, x) -> schedule(this, sweepPeriod));
+                _sweeper.asyncSweep().whenComplete((r, x) -> schedule(this, sweepPeriod));
             }
         }, sweepPeriod);
```

Comment:
While we're improving the sweeping, let's add getter/setter for `sweepPeriod`, and use the getter here so that it can be changed by JMX and the change will be reflected in the next sweep cycle.

Reply:
Done.

Reply:
Since we're now allowing these two config settings to be changed via the setter, I've also added some validation code to the setter to avoid negative/zero number of threads or negative periods.